### DISCUSSION TITLE
Why you place b43 firmware in /lib/firmware rather than install firmware installer package

### DISCUSSION THREAD

**Discussion by CicadaSeventeen** [SpiralLinux/SpiralLinux-project#167] [Q&A]
`b43` firmware is not installed by any packages. I even can not find them on kernel.org.
Debian provides installer for them in repository. However, SpiralLinux does not use it but to place it in `chroot` directory.
Is there any reason to do so? Why not use installer packages?

**Comment by geckolinux** [comment]
What Debian provides is just an installer script that downloads the firmware from https://www.lwfinger.com/b43-firmware/broadcom-wl-6.30.163.46.tar.bz2 , and it also depends on `b43-fwcutter` that extracts the firmware files and places them in in `/lib/firmware/b43/`. We can try to include it in the package list, but I seem to remember that it failed during the `live-build` stage in the chroot environment.

**Reply by CicadaSeventeen** [reply]
Thanks. BTW, why you put 3 packages in the files. If just use packages from repository to build the iso, will there be any problem?

**Reply by geckolinux** [reply]
> BTW, why you put 3 packages in the files

Those are necessary to override the packages that would normally come from the Debian Stable repositories. There is actually a 4th package in the most recent version, but it was too big to upload to Github so I didn't include it in the tar.gz .

- `firmware-iwlwifi_20210818-1_all.deb` - https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=987116
- `broadcom-sta-dkms_6.30.223.271-20~bpo11+1_all.deb` - https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1012647
- `r8168-dkms_8.050.03-1~bpo11+1_all.deb` - https://groups.google.com/g/linux.debian.bugs.dist/c/7SVtJBW1uLA
- `calamares_3.2.56-1~bpo11+1_amd64.deb` - A newer version of Calamares is required for certain functions in SpiralLinux.

**Reply by CicadaSeventeen** [reply]
Thanks a lot. The first package is from Sid right?
